### PR TITLE
Separating manager yaml files

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fgelcer** [comment]
@ShlomiBalalis , what is the motivation behind this change?
All new yaml files are exactly the same, where only the debian9 yaml file has an extra parameter.
In this case, shouldn't we be using a "base yaml" for all manager jobs, and then specific yaml for the jobs are changing or adding new parameters?

**Comment by ShlomiBalalis** [comment]
@fgelcer I figured it will be more orderly to separate them one and for all

**Comment by fgelcer** [comment]
> @fgelcer I figured it will be more orderly to separate them one and for all

i agree that create separated files is a good idea, but in this case, we have (right now) 4 copies of the exactly same file, and only Debian 9 has an extra parameter..

File: jenkins-pipelines/manager-backup.jenkinsfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ longevityPipeline(
     ip_ssh_connections: 'public',
     aws_region: '''["us-east-1", "us-west-2"]''',
     test_name: 'mgmt_cli_test.MgmtCliTest.test_backup_feature',
-    test_config: 'test-cases/manager/manager-regression-multiDC-set-distro.yaml',
+    test_config: 'test-cases/manager/manager-regression-centos-multiDC-set-distro.yaml',
```
Comment (fruch) [inline jenkins-pipelines/manager-backup.jenkinsfile:13]: why did you rename this file ?
the idea was more like:
```
test_config: ['test-cases/manager/manager-regression-multiDC-set-distro.yaml', 'configurations/centos7.yaml']
```
and then the platform specific files are just the parameter needed to be overwritten.

File: test-cases/manager/manager-regression-debian9-multiDC-set-distro.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+stress_cmd: "cassandra-stress write cl=QUORUM n=1200300 -schema 'replication(strategy=NetworkTopologyStrategy,us-eastscylla_node_east=2,us-west-2scylla_node_west=1)' -port jmx=6868 -mode cql3 native -rate threads=200 -pop seq=400200300..600200300"
```
Comment (fruch) [inline test-cases/manager/manager-regression-debian9-multiDC-set-distro.yaml:1]: I agree with @fgelcer the idea wasn't to duplicate the test yamls, but to create extra yamls for the overrides

File: vars/managerPipeline.groovy [hunk 3]
```diff
@@ -158,10 +158,6 @@ def call(Map pipelineParams) {
                                         export SCT_SCYLLA_MGMT_PKG="${params.scylla_mgmt_pkg}"
                                     fi
 
-                                    if [[ ! -z "${params.aws_root_disk_name_monitor}" ]] ; then
```
Comment (fruch) [inline vars/managerPipeline.groovy:161]: while remove here params from the pipeline, we should remove most of them from here, and leave only the one needed to be called from other jobs
for example, all those can be configure in a per platform yaml, and can be remove from the pipeline params:
```
params.ami_id_monitor
params.ami_monitor_user
params.scylla_repo_m
```

File: jenkins-pipelines/manager/ubuntu18-manager-sanity.jenkinsfile [hunk 4]
```diff
@@ -10,7 +10,7 @@ managerPipeline(
     ip_ssh_connections: 'public',
     aws_region: '''["us-east-1", "us-west-2"]''',
     test_name: 'mgmt_cli_test.MgmtCliTest.test_manager_sanity',
-    test_config: 'test-cases/manager/manager-regression-multiDC-set-distro.yaml',
+    test_config: 'test-cases/manager/manager-regression-ubuntu18-multiDC-set-distro.yaml',
     ami_id_monitor: 'ami-08b314ce48a790a19', // Canonical, Ubuntu, 18.04 LTS, amd64 bionic image build on 2019-07-22
     ami_monitor_user: 'ubuntu',
```
Comment (fruch) [inline jenkins-pipelines/manager/ubuntu18-manager-sanity.jenkinsfile:15]: those should be moved into `ubuntu18.04.yaml`

File: jenkins-pipelines/manager/ubuntu16-manager-sanity.jenkinsfile [hunk 5]
```diff
@@ -10,7 +10,7 @@ managerPipeline(
     ip_ssh_connections: 'public',
     aws_region: '''["us-east-1", "us-west-2"]''',
     test_name: 'mgmt_cli_test.MgmtCliTest.test_manager_sanity',
-    test_config: 'test-cases/manager/manager-regression-multiDC-set-distro.yaml',
+    test_config: 'test-cases/manager/manager-regression-ubuntu16-multiDC-set-distro.yaml',
     ami_id_monitor: 'ami-092d0d014b7b31a08', // Canonical, Ubuntu, 16.04 LTS, amd64 xenial image build on 2019-02-12
     ami_monitor_user: 'ubuntu',
     scylla_repo_m: 'https://s3.amazonaws.com/downloads.scylladb.com/deb/ubuntu/scylla-2019.1-xenial.list',
```
Comment (fruch) [inline jenkins-pipelines/manager/ubuntu16-manager-sanity.jenkinsfile:16]: those should be moved into `ubuntu16.04.yaml`

File: jenkins-pipelines/manager/debian9-manager-sanity.jenkinsfile [hunk 6]
```diff
@@ -10,10 +10,9 @@ managerPipeline(
     ip_ssh_connections: 'public',
     aws_region: '''["us-east-1", "us-west-2"]''',
     test_name: 'mgmt_cli_test.MgmtCliTest.test_manager_sanity',
-    test_config: 'test-cases/manager/manager-regression-multiDC-set-distro.yaml',
+    test_config: 'test-cases/manager/manager-regression-debian9-multiDC-set-distro.yaml',
     ami_id_monitor: 'ami-0cfac3931b2a799d1', // Debian stretch image
     ami_monitor_user: 'admin',
-    aws_root_disk_name_monitor: '/dev/xvda',
     scylla_repo_m: 'https://s3.amazonaws.com/downloads.scylladb.com/deb/debian/scylla-2019.1-stretch.list',
```
Comment (fruch) [inline jenkins-pipelines/manager/debian9-manager-sanity.jenkinsfile:16]: those should be moved into `debian9.yaml`

File: jenkins-pipelines/manager-backup.jenkinsfile (hunk 1 above)
Comment (ShlomiBalalis) [reply jenkins-pipelines/manager-backup.jenkinsfile:13]: I did not know this was possible, I'll change them accordingly

File: vars/managerPipeline.groovy (hunk 3 above)
Comment (fgelcer) [reply vars/managerPipeline.groovy:161]: the idea of having those in the pipeline, was to git the ability to the dev to put here any manager he wants (as it is running on the monitor node) and be able to set the specific `user` and the `back end scylla` used by the `manager`

**Comment by fgelcer** [comment]
@ShlomiBalalis , isn't it missing now the centos yaml and pipeline files?

File: vars/managerPipeline.groovy (hunk 3 above)
Comment (ShlomiBalalis) [reply vars/managerPipeline.groovy:161]: @fgelcer So should I leave it there?

Comment (fruch) [reply vars/managerPipeline.groovy:161]: @fgelcer can you point me to where those pipeline params are use in Jenkins.... I doubt they were ever used

Comment (fgelcer) [reply vars/managerPipeline.groovy:161]: i don't think we need to leave it

**Comment by fgelcer** [comment]
> @ShlomiBalalis , isn't it missing now the centos yaml and pipeline files?

what about this one, @ShlomiBalalis ?

**Comment by ShlomiBalalis** [comment]
> > @ShlomiBalalis , isn't it missing now the centos yaml and pipeline files?
> 
> what about this one, @ShlomiBalalis ?

Centos doesn't have any extra settings, so there's no need for an extra config file

**Comment by roydahan** [comment]
@fruch did you approve this?
We need the merge and backports ASAP.
